docs/posts/ibis-analytics/index.qmd

```diff
@@ -1,341 +1,1969 @@
-
-I've experienced many pains of the ecosystem myself. I downloaded a 1 GB CSV from a database GUI and used pandas to munge it because I couldn't figure it out in SQL. When building larger ML demos that included data preprocessing on GBs, I hit OOM errors and learned about how much data expands in memory and why to store files in compressed parquet and how to partition data. I started using PySpark and Dask to overcome hurdles. I slowly learned about databases. Fortunately while I've been learning, others ([including many Ibis contributors](https://github.com/ibis-project/ibis/contributors)) have been building a modern, open-source Python data ecosystem!
-
-Ibis plays a part but is only one component in any real data project. We'll look at how to combine a variety of open-source tools and freemium services including:
+As a Python data user, I've wanted a more modular, composable, and scalable
+ecosystem. *I think it's here*. Wes McKinney released pandas c. 2009 to bring
+dataframes into Python and it became one of the most used software packages. It
+was built when small data was smaller and though still an excellent dataframe
+library today has some downsides [Wes wrote about in his "Apache Arrow and the
+'10 things I hate about pandas'" blog
+post](https://wesmckinney.com/blog/apache-arrow-pandas-internals). Wes created
+Ibis (and co-created Apache Arrow) to address these issues and more with a
+different approach than any other Python dataframe library by decoupling the
+dataframe API from the backend implementation. This allows Ibis to support 20+
+backends today, including pandas and DuckDB and Polars and Postgres and BigQuery
+and Snowflake and PySpark and over a dozen more.
+
+I've experienced many pains of the ecosystem myself. I've downloaded a 1 GB CSV
+from a database GUI and used pandas to munge it because I couldn't figure it out
+in SQL. When building larger ML demos that included data preprocessing on GBs, I
+hit OOM errors and learned about how much data expands in memory and why to
+store files in compressed Parquet. I learned about partitioning strategies and
+the joy of subtle differences and incompatibilities across data system. I
+started using PySpark and Dask to overcome hurdles. I slowly learned about
+databases. Fortunately while I've been learning, others ([including many Ibis
+contributors](https://github.com/ibis-project/ibis/contributors)) have been
+building a modern, open-source Python data ecosystem! Now I can just use what
+they've built.
+
+In this blog, we'll look at how to build an end-to-end analytics project using
+Ibis as the frontend for data in Python. Ibis plays a large part, but is only
+one component in any real data project.
+
+We'll combine a variety of open-source tools and freemium services including:
```

Comment:
It could be useful to partition this into two groups: software components and data, since the data are software-related.

Reply:
I like the big list